lib/OpenQA/Worker.pm

```diff
@@ -324,6 +324,13 @@ sub init {
             });
     }
 
+    # Store the packages list of the environment the worker runs in
+    if (my $cmd = $global_settings->{PACKAGES_CMD}) {
+        my $pool_directory = $self->pool_directory;
```

Comment:
You've now moved the code to the worker startup as I suggested. That's good but I don't think this file will survive the cleanup of the pool directory this way. The `OpenQA::Worker::_clean_pool_directory` function has an exclude list anyways so you could simple extend it.

Reply:
Well, this is not necessarily a better place. There is no guarantee that there are no package changes since the worker start so we might need to collect the packages and versions at start of a job hoping that this is the best approximation. We can argue that it's the admins job to ensure this consistency. I guess this new place is fine.

Reply:
I know but I would also claim it is the admin's job to ensure consistency.

And when it comes to the worker and worker cache service itself: Not the installed version will be used but the version which was installed at the time of the last service restart.

(It is also questionable whether it is a good idea to update os-autoinst while isotovideo is executed. It would be better to stop all workers, do the updates and start the workers again.)

Reply:
I think the process to list the packages doesn't consume too much time : real 0m1.342s and I don't think will be a big deal in other OS.

Why don't we keep it before job execution to be sure that the conditions are exactly what we are storing. The only problem could be if we have concurrent jobs for the same pool (rewriting the package list file). But I guess is not the case.

Reply:
I'm torn so let's @okurz decide.

One further note which speaks for executing it before every job: It is an optional setting so local testing wouldn't be affected anyways. On the production instance an additional delay of ~ 2 seconds before starting a job might be acceptable.

> The only problem could be if we have concurrent jobs for the same pool (rewriting the package list file).

That can not be the case (unless the admin starts two worker services sharing the same pool directory which is not supported at all).

> to be sure that the conditions are exactly what we are storing?

There are still caveats with that, see my last comment.

Reply:
Let's keep it as you have it now, i.e. Start of worker

Reply:
Since we're keeping it now at the start of the worker you need to address the point from my first comment.